.github/workflows/ci-workflow.yml

```diff
@@ -0,0 +1,92 @@
+name: Build, test and package esque
+on:
+  push:
+    branches:
+      - 61-installation-and-packaging
+  pull_request:
+    branches:
+      - 61-installation-and-packaging
+
+jobs:
+  test-style:
+    name: Code style check on ${{ matrix.os }} with Python ${{ matrix.python }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        python: ['3.x', '3.6', '3.7', '3.8', 'pypy3' ]
+        os: ['ubuntu-latest']
+    steps:
+      - uses: actions/checkout@v1
+      - name: Setup python
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python }}
+          architecture: x64
+        run: |
+          pre-commit install --install-hooks
+      - name: Run pre-commit hooks
+        run: pre-commit run
+  test-integration:
+    name: Integration test with Python ${{ matrix.python }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python: ['3.x', '3.6', '3.7', '3.8', 'pypy3' ]
+      max-parallel: 1
+    steps:
+      - uses: actions/checkout@v1
+      - name: Setup python and install packages
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python }}
+          architecture: x64
+        run: |
+          sudo apt-get install -y python3-setuptools python3-virtualenv python3-pip virtualenv python3-venv
+          pip3 install pre-commit
+      - name: Install and configure poetry
+        run: |
```

Comment:
I'd rather use `pip3 install "poetry==1.0.0b3"` so we know which version we get.